api/provider/aws/environment_update.go

```diff
@@ -33,6 +39,93 @@ func (e *EnvironmentProvider) Update(req models.UpdateEnvironmentRequest) error
 		}
 	}
 
+	if req.Links != nil {
+		sourceEnvSG, err := readSG(e.AWS.EC2, getEnvironmentSGName(fqEnvironmentID))
+		if err != nil {
+			return err
+		}
+
+		// check which environments are currently linked
+		currentTags, err := e.TagStore.SelectByTypeAndID("environment", environmentID)
+		if err != nil {
+			return err
+		}
+
+		currentLinkedEnvIDs := []string{}
+		for _, tag := range currentTags.WithKey("link") {
+			currentLinkedEnvIDs = strings.Split(tag.Value, ",")
+		}
+
+		checkForAWSError := func(err error) error {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "DoesNotExist" {
+				log.Printf("[WARN] skipping environment link/unlink since security group '%s' does not exist\n", getEnvironmentSGName(fqEnvironmentID))
+				return nil
+			}
+
+			return err
+		}
+
+		contains := func(value string, values []string) bool {
```

Comment:
this should probably be a standalone somewhere else. Maybe bottom of this file or in `helpers.go`

Reply:
Moved to `helpers.go` ✅